src/sqlfluff/core/rules/std/L010.py

```diff
@@ -169,12 +169,13 @@ def _eval(self, segment, memory, parent_stack, **kwargs):
             )
             return LintResult(
                 anchor=segment,
-                fixes=[
-                    LintFix(
-                        "edit",
-                        segment,
-                        segment.__class__(raw=fixed_raw, pos_marker=segment.pos_marker),
-                    )
-                ],
+                fixes=[self._get_fix(segment, fixed_raw)],
                 memory=memory,
             )
+
+    def _get_fix(self, segment, fixed_raw):
```

Comment:
Introducing this new overridable function allows rule L030, which handles function names, to continue working even though a parsed `function_name` now includes an additional level.

Reply:
Can you add a note to that effect in the docstring for `_get_fix()`. Logic makes sense, but I assume future versions of ourselves will forget.

Reply:
Will do. I'm surprised the linter didn't flag this.